arctic/arctic.py

```diff
@@ -202,6 +213,44 @@ def list_libraries(self):
                         libs.append(coll[:-1 * len(self.METADATA_COLL) - 1])
         return libs
 
+    @mongo_retry
+    def invalidate_cache(self):
+        """
+        Deletes the data in the collection used to cache the result
+        of list_libraries
+        :return:
+        """
+        self._conn.meta_db.cache.drop()
+
+    @mongo_retry
+    def _list_libraries_cached(self):
+        """
+        Returns
+        -------
+        List of Arctic library names from a cached collection (global per mongo cluster) in mongo.
+        Long term list_libraries should have a use_cached argument.
+        TODO: Maybe convert this to a decorator so this code can be used in a generic way.
+        """
+        mongo_db = self._conn.meta_db
+        col_name = 'cache'
+        if col_name not in mongo_db.collection_names():
+            mongo_db.create_collection(col_name).create_index("date", expireAfterSeconds=3600)
+
+        cache_col = mongo_db[col_name]
+        coll_data = cache_col.find_one({"type": "list_libraries"})
+        if coll_data:
+            logger.debug('Library names are in cache.')
+            return coll_data['data']
+        library_names = self._list_libraries()
+        # TODO: Check if ret is serializable to BSON by mongo. For now it's a list of strings.
+        cache_col.update_one(
```

Comment:
Assumes write access to this database which most users won't have => lead to an exception? Is it worth separating out the cache update from the usage?

Reply:
Yeah, it will throw an exception but will be caught in the list_libraries code and default to the slow path. I am not sure if it will be trivial to schedule it in the background. An option is to regenerate it on initialize_library and have a very high expiry time, but that slows initialize library every time.